### PR TITLE
add gulp task to convert markdown to html

### PR DESCRIPTION
**Pull request by mmalerba**
additionally replaces comments in the outputted html of the form `<!-- example(label) -->` with `<div example="label">`

### CODE REVIEW

File: tools/gulp/tasks/docs.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import gulp = require('gulp');
+const markdown = require('gulp-markdown');
+const transform = require('gulp-transform');
+
+
+const EXAMPLE_PATTERN = /<!--\W*example\(([^)]+)\)\W*-->/g;
```
Comment (jelbourn) [inline tools/gulp/tasks/docs.ts]: Can you add a comment explaining what we're replacing and why we're replacing it?

File: tools/gulp/tasks/docs.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+
+gulp.task('docs', () => {
+  return gulp.src(['src/lib/**/*.md'])
+      .pipe(markdown())
+      .pipe(transform((content: string) =>
+          content.toString().replace(EXAMPLE_PATTERN, (match: string, name: string) =>
+              `<div example="${name}"></div>`)))
```
Comment (jelbourn) [inline tools/gulp/tasks/docs.ts]: How about `material-docs-example` to be super explicit?

**Comment by mmalerba** [comment]
Done

File: tools/gulp/tasks/docs.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import gulp = require('gulp');
+const markdown = require('gulp-markdown');
+const transform = require('gulp-transform');
+
+
+// Our docs contain comments of the form `<!-- example(...) -->` which serve as placeholders where
+// example code should be inserted. We replace these comments with divs that have a
+// `material-docs-example` attribute which can be used to locate the divs and initialize the example
+// viewer.
```
Comment (tinayuangao) [inline tools/gulp/tasks/docs.ts]: very nit: do you want to comment on naming convention for $name?

Comment (mmalerba) [inline tools/gulp/tasks/docs.ts]: At some point yes, I'm not really sure what's going to go there yet...

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**